torch/csrc/lazy/core/shape.cpp

```diff
@@ -30,11 +35,79 @@ size_t Shape::numel() const {
 
 hash_t Shape::hash(bool bakeInSizes) const {
   if (bakeInSizes) {
-    return HashCombine(Hash(scalar_type_), DataHash(sizes_.data(), sizes_.size() * sizeof(int64_t)));
+    return HashCombine(
+        Hash(scalar_type_),
+        DataHash(sizes_.data(), sizes_.size() * sizeof(int64_t)));
   } else {
     return HashCombine(Hash(scalar_type_), Hash(sizes_.size()));
   }
 }
 
-}  // namespace lazy
-}  // namespace torch
+c10::SymbolicShape Shape::get_symbolic_shape() const {
+  if (!is_symbolic_) {
+    return c10::SymbolicShape();
+  }
+  TORCH_INTERNAL_ASSERT(
+      sizes_.size() == is_symbolic_->size(),
+      "Dims of two values are not consistent");
+  std::vector<c10::optional<int64_t>> symbolic_dims;
+  for (int64_t i = 0; i < sizes_.size(); i++) {
+    if (is_symbolic_->at(i)) {
+      symbolic_dims.emplace_back(c10::nullopt);
+    } else {
+      symbolic_dims.emplace_back(sizes_.at(i));
+    }
+  }
+  return c10::SymbolicShape(symbolic_dims);
+}
+
+void Shape::set_from_symbolic(c10::SymbolicShape& ss) {
+  is_symbolic_ = ss.concreteDims();
```

Comment:
isn't the logic inverted here? `concreteDims` returns dimensions that are static and is_symbolic_ seems to imply we want the opposite?

Reply:
oops, good catch